src/Nethermind/Nethermind.Core/Collections/CollectionExtensions.cs

```diff
@@ -22,5 +26,17 @@ public static void AddRange<T>(this ICollection<T> list, params T[] items)
                 list.Add(items[index]);
             }
         }
+
+        public static void NoResizeClear<TKey, TValue>(this ConcurrentDictionary<TKey, TValue> dict)
+                where TKey : notnull
+        {
+            // We iterate over the keys and remove them one by one because calling Clear() on
+            // the ConcurrentDictionary resets its capacity to 31 and then it has to constantly resize.
+            foreach (TKey key in dict.Keys)
+            {
+                dict.TryRemove(key, out _);
```

Comment:
Wouldn't it be better to grab the internals and clear the `_buckets` ourselves?

Reply:
Looks horrible to do because of the private types involved 😢

Really just need an option on .Clear to not resize back to `HashHelpers.GetPrime(31)` will raise issue in runtime

```csharp
Tables tables = _tables;
var newTables = new Tables(new VolatileNode[HashHelpers.GetPrime(DefaultCapacity)], tables._locks, new int[tables._countPerLock.Length], tables._comparer);
_tables = newTables;
_budget = Math.Max(1, newTables._buckets.Length / newTables._locks.Length);
```

Reply:
https://github.com/dotnet/runtime/issues/107016

Reply:
Actually...

Reply:
Done